packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -336,6 +340,15 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
   const defaultOnRenderVideoTile = useCallback(
     (participant: VideoGalleryRemoteParticipant, isVideoParticipant?: boolean) => {
       const remoteVideoStream = participant.videoStream;
+      // Async call to retrieve custom data model information to replace display name in remote tiles with camera on.
+      if (onFetchAvatarPersonaData) {
+        (async () => {
+          const newPaticipantData = await onFetchAvatarPersonaData(participant.userId);
+          participant.displayName = newPaticipantData.text;
```

Comment:
this mutates the participant displayName prop which is bad practice (if this is used elsewhere the value of the property is suddenly changed).
Instead we'll need to call a setState(participantsMergedWithFetchedData) once this returns with the new data. This also means we should have no name until the name is fetched from the onFetchAvatarPersonaData

Reply:
Without a useState or similar there's also nothing to tell this component to rerender when this async block completes. (imagine it took 10 seconds to complete and everything in the app was stable and unchanging -- nothing here would tell the component to rerender to update the name)